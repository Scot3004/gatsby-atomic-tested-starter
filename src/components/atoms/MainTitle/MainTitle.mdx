---
name: Main Title
---

import { Playground } from 'docz';
import MainTitle from '.';

# Main Title

Renders a main title

The color could be set by a theme with `primaryColor` using the context API

## Playground

<Playground>
  <MainTitle theme={{primaryColor: '#0ad'}}>Main title</MainTitle>
</Playground>
