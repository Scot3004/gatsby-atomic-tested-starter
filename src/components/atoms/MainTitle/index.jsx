import styled from 'styled-components';

export const MainTitle = styled.h1`
  color: #663399;
`;

export default MainTitle;
