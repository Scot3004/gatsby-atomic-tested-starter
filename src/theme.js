const theme = {
  backgroundColor: '#f5f5f5',
  textColor: '#000',
  mutedColor: '#555',
  primaryColor: '#663399',
  visitedColor: '#402060'
};

export default theme;
