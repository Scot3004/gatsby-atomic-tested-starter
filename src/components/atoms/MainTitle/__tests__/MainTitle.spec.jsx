import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme from '../../../../theme';
import MainTitle from '..';

describe('When I visit the site', () => {
  const tree = renderer.create(<MainTitle theme={theme} />).toJSON();
  test('Then the site should have a general title', () => {
    expect(tree).toMatchSnapshot();
  });
});
