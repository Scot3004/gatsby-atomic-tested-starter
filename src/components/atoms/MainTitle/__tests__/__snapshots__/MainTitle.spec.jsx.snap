// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I visit the site Then the site should have a general title 1`] = `
.c0 {
  color: #663399;
}

<h1
  className="c0"
/>
`;
